scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/listener/JobTaskStatusListener.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.scheduler.common.listener;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import org.objectweb.proactive.annotation.ImmediateService;
+import org.objectweb.proactive.api.PAActiveObject;
+import org.ow2.proactive.scheduler.common.NotificationData;
+import org.ow2.proactive.scheduler.common.Scheduler;
+import org.ow2.proactive.scheduler.common.SchedulerEvent;
+import org.ow2.proactive.scheduler.common.SchedulerEventListener;
+import org.ow2.proactive.scheduler.common.job.*;
+import org.ow2.proactive.scheduler.common.task.TaskInfo;
+import org.ow2.proactive.scheduler.common.task.TaskStatus;
+
+
+public class JobTaskStatusListener implements SchedulerEventListener, Serializable {
+    public static final int WAIT_TIME = 1000;
+
+    private List<TimestampedData<JobInfo>> jobEvents = new CopyOnWriteArrayList<>();
+
+    private List<TimestampedData<TaskInfo>> taskEvents = new CopyOnWriteArrayList<>();
+
+    public static JobTaskStatusListener attachListener(Scheduler scheduler) throws Exception {
+        JobTaskStatusListener listener = new JobTaskStatusListener();
+        listener = PAActiveObject.turnActive(listener);
+        scheduler.addEventListener(listener, false);
+        return listener;
+    }
+
+    @Override
+    public void jobStateUpdatedEvent(NotificationData<JobInfo> notificationData) {
+        final JobInfo jobInfo = notificationData.getData();
+        final TimestampedData<JobInfo> timestampedData = new TimestampedData<>(jobInfo);
+        jobEvents.add(timestampedData);
+    }
+
+    @Override
+    public void taskStateUpdatedEvent(NotificationData<TaskInfo> notificationData) {
+        final TaskInfo taskInfo = notificationData.getData();
+        final TimestampedData<TaskInfo> timestampedData = new TimestampedData<>(taskInfo);
+        taskEvents.add(timestampedData);
+    }
+
+    @ImmediateService
```

Comment:
Is the ImmediateService really applicable here ?

Reply:
Without ImmediateService, after it becomes AO, I cannot be blocking until I receive some event and actually receiving it.